### DISCUSSION TITLE
Nuxt 3 client side plugin

### DISCUSSION THREAD

**Discussion by iamsaintcyr** [nuxt/nuxt#24226] [Questions] [answered]
Hello,
The documentation for plugins says this:

> Nuxt will automatically read the files in your plugins directory and load them. You can use .server or .client suffix in the file name to load a plugin only on the server or client side.

I've added a plugin with the name "hello.client.ts" in which I use a package that has methods like "info", "success", etc. 
This is how I use it in my pages: 

```
const { $hello } = useNuxtApp();
$hello.success('It works');
```
However, when I refresh the page I get the following error:

`Cannot read properties of undefined (reading 'success') `
I have to do this for it to work:
```
if(process.client){
  $hello.success('You did it!');
}
```

How can I please fix this?
I am using Nuxt 3 with Composition API(setup) and Typescript

**Comment by huang-julien** [comment]
hello can you provide a minimal reproduction please ? you can use stackblitz so it's easier for us to load your project and debug it

**Reply by iamsaintcyr** [reply, marked as answer]
It's good now !

I forgot to use it into a lifecycle hook.
Btw, this is the stackblitz link: https://stackblitz.com/edit/nuxt-starter-uzjusu?file=pages%2Findex.vue

Thanks a lot !
